pkg/middleware/org_redirect.go

```diff
@@ -44,7 +44,13 @@ func OrgRedirect(cfg *setting.Cfg) web.Handler {
 			return
 		}
 
-		newURL := fmt.Sprintf("%s%s?%s", cfg.AppURL, strings.TrimPrefix(c.Req.URL.Path, "/"), c.Req.URL.Query().Encode())
+		var queryString string = c.Req.URL.Query().Encode()
+		if strings.Contains(queryString, "kiosk=") && !strings.Contains(queryString, "kiosk=tv") {
```

Comment:
I would suggest to better use [Values.Has()](https://pkg.go.dev/net/url#Values.Has) for checking if. `kiosk` query parameter exists.
Also I would copy `c.Req.URL.Query()` to a variable, delete the `kiosk` parameter (using [Values.Del()](https://pkg.go.dev/net/url#Values.Del), encode it and then append the `kiosk`.